core/src/main/scala/org/bykn/bosatsu/Declaration.scala

```diff
@@ -469,57 +479,69 @@ object Declaration {
       }
   }
 
-  val lambdaP: Indy[Lambda] = {
+  def lambdaP(parser: Indy[Declaration]): Indy[Lambda] = {
     val params = Indy.lift(P("\\" ~/ maybeSpace ~ Pattern.bindParser.nonEmptyList))
 
     Indy.blockLike(params, parser, P(maybeSpace ~ "->"))
       .region
       .map { case (r, (args, body)) => Lambda(args, body.get)(r) }
   }
 
-  def matchP(expr: Indy[Declaration]): Indy[Match] = {
-    val withTrailing = expr <* Indy.lift(maybeSpace)
-    val branch = Indy.block(Indy.lift(Pattern.matchParser), withTrailing)
+  def matchP(arg: P[NonBinding], expr: Indy[Declaration]): Indy[Match] = {
```

Comment:
should be `Indy[NonBinding]` since we are not applying the indentation, and wrapped `Parens` nodes would need to know the indentation.